pyvista/plotting/renderers.py

```diff
@@ -362,6 +362,8 @@ def set_active_renderer(self, index_row, index_column=None):
             self._active_index = index_row
             return
 
+        if index_column is None:
+            raise IndexError(f'Only one index is given for ({self.shape}).')
```

Comment:
Allowing `index_column=None` suggests that `index_column` is optional, but raising an error here suggests that it's actually required. So why even allow it to be optional / have a default value of `None`? Maybe it's better to do:
``` diff
- def set_active_renderer(self, index_row, index_column=None):
+ def set_active_renderer(self, index_row, index_column):
```
That way a `TypeError` is automatically raised, e.g. 
```TypeError: set_active_renderer() missing 1 required positional argument: 'index_column```'

Even better might be to add type annotations:

``` diff
- def set_active_renderer(self, index_row, index_column=None):
+ def set_active_renderer(self, index_row: int, index_column: int):
```

And let `mypy` find any issues with the code, e.g. invalid less-than comparisons with `None`.

Reply:
I guess it's optional since the shape may be one-dimensional... But in the 1-d case no checks are made to ensure that (1) the `row_index` is not out of range and (2) the `column_index` is actually `None`.

So maybe overall the validation logic needs to be updated, maybe something like:

``` python
from pyvista.core import _validation

# validate row index
_validation.check_instance(index_row, int)
if index_row < 0 or index_row >= self.shape[0]:
    raise IndexError(f'Row index is out of range ({self.shape[0]})')

# validate column index and set active index
if len(self.shape) == 1:
    _validation.check_type(index_column, type(None))
    self._active_index = index_row
else:
    _validation.check_instance(index_column, int)
    if index_column < 0 or index_column >= self.shape[1]:
        raise IndexError(f'Column index is out of range ({self.shape[1]})')
    self._active_index = self.loc_to_index((index_row, index_column))


```

Reply:
> I guess it's optional since the shape may be one-dimensional...

This is my understanding too. However, I'm not fully confident because there aren't any failing tests with the current PR. The logic is very opaque, so to make my original comment more precise.  I suggest that we figure out the branches of logic that result in different kinds of shape for the Plotter object, and see if there are any tests for it.  Maybe there aren't and we need to add some.

I suspect your suggested code is in the right direction to fix this.